Use `getError()` for Guzzle CurlException

ElliotFriend opened the pull request:
Issue #124

The `getResponse()` method is not a valid method for Guzzle's
CurlException class. `getError()` seems to be doing the trick,
though.

Further, the `getError()` method doesn't have a `getBody()` member
function. It just returns the error.

Signed-off-by: Elliot Voris elliot@voris.me

mikevanwinkle commented:
@ElliotFriend this conflict with this change that I just merged, which I believe is similar #83. Can you take a look and update your pull request if it's still needed?

ElliotFriend commented:
I just saw that, and I'll give it a check now. Thanks!

ElliotFriend commented:
Under the same conditions (no connection to the internet), I'm seeing a brand new error:

``` bash
elliot@ransford ~/Development/Pantheon/cli $ bin/terminus auth login elliot@voris.me
Your dashboard password (input will not be shown): 
Logging in as elliot@voris.me
PHP Warning:  vsprintf(): Too few arguments in /home/elliot/Development/Pantheon/cli/php/class-terminus.php on line 204
Error: 
```

ElliotFriend commented:
Although, from what I can tell, both `$params` and `$message` are both set. I found out that the `Terminus::error` call that I'm catching is line 164 here: https://github.com/pantheon-systems/cli/blob/master/php/class-terminus-command.php#L164

ElliotFriend committed:
`Terminus::error` only accepts two arguments, three given

`Terminus::error` was not working properly. Wrapping the request
and the error message into an array made it happy.

Signed-off-by: Elliot Voris <elliot@voris.me>

ElliotFriend commented:
I rebased from master, as the changes provided by @jbutkus were a lot better than mine. Then, `Terminus::error` started giving me an error.

The pull request now reflects a fix to that issue.
